rdrf/rdrf/users/views.py

```diff
@@ -0,0 +1,48 @@
+import logging
+
+from django.contrib import messages
+from django.core.exceptions import NON_FIELD_ERRORS, ValidationError
+from django.shortcuts import render, redirect
+from django.utils.translation import gettext_lazy as _
+from django.views import View
+
+from rdrf.users.forms import EmailChangeForm
+from rdrf.users.utils import sync_user_email_update
+
+logger = logging.getLogger(__name__)
+
+
+class ChangeEmailAddressView(View):
+
+    @staticmethod
+    def _render_template(request, user, form):
+
+        context = {
+            'current_username': user.username,
+            'current_email': user.email,
+            'patient': user.patient,
+            'form': form
+        }
+
+        return render(request, 'user/change_email_address.html', context=context)
+
+    def get(self, request):
+        user = request.user
+        form = EmailChangeForm(user=request.user)
+
+        return self._render_template(request, user, form)
+
+    def post(self, request):
+        user = request.user
+        form = EmailChangeForm(request.POST, user=user)
+
+        if form.is_valid():
+            sync_user_email_update(user, form.cleaned_data['new_email'])
+            messages.add_message(self.request,
+                                 messages.SUCCESS,
+                                 f'{_("Email address has been updated successfully for user")}: {user.get_full_name()}')
+            return redirect('email_address_change')
```

Comment:
I think having a new EmailNotification to send on successful change would be useful to curators, and to notify the user themselves at their old email.

Reply:
Any tips on what to say in this email notification?

i.e. something like...

> You're receiving this email because your recent request to change your email address has been completed. This is a courtesy message to let you know that this email address is no longer active. You can request to change your email address at any time by using the "Change Email Address" available from your menu.

Reply:
Yeah that's good. I would just be clear that it's coming from 'the registry' so they don't think it's from their email provider.

Reply:
Done.